### ISSUE TITLE
Add deterministic unit test for LCAO gaussian basis with k-points

### ISSUE THREAD

**Issue by prckent** [QMCPACK/qmcpack#2012]
Per discussion in #1992 @mcbennet will add this. Thanks!

**Comment by ye-luo** [comment]
@mcbennet For creating unit test, QMCWaveFunctions/tests/test_MO.cpp is a good example. It was made for open boundary condition.

**Comment by mcbennet** [comment]
Thank you! 

For the tests, I am planning to compare the MOs along a particular path in real space between qmcpack and pyscf. Alternatively, the code that I wrote  which Anouar and I were using in Argonne (and also prints the MOs along a path) could be ported and folded into the test. But would it not be better to simply print the MOs directly from pyscf to compare with those from qmcpack?

**Comment by ye-luo** [comment]
I would prefer saving a list of values from pyscf to the test source code. Then compare qmcpack numbers directly to them. Keep the pyscf script next to the test so that someone can reproduce it in the future if needed.

**Comment by prckent** [comment]
I agree with @ye-luo . Not an issue with saved values, but also bear in mind that we can't assume that PySCF will be available on the system where QMCPACK is installed and run. e.g. It is not yet installed on oxygen where we do most of our testing... (will fix soon).

**Comment by mcbennet** [comment]
Noted. PR should sent out soon.
